library/src/main/java/io/ona/kujaku/utils/FeatureFilter.java

```diff
@@ -0,0 +1,152 @@
+package io.ona.kujaku.utils;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.mapbox.geojson.Feature;
+import com.mapbox.geojson.FeatureCollection;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import io.ona.kujaku.comparisons.Comparison;
+import io.ona.kujaku.comparisons.EqualToComparison;
+import io.ona.kujaku.comparisons.RegexComparison;
+
+/**
+ * Created by Ephraim Kigamba - ekigamba@ona.io on 14/02/2019
+ */
+
+public class FeatureFilter {
+
+    private Builder builder;
+
+    private FeatureFilter(@NonNull Builder builder) {
+        this.builder = builder;
+    }
+
+    public FeatureCollection filter() {
+        List<Feature> featuresList = builder.getFeatureCollection().features();
+        ArrayList<Feature> filteredFeatures = new ArrayList<>();
+
+        HashMap<String, Comparison> comparisons = new HashMap<>();
+
+        EqualToComparison equalToComparison = new EqualToComparison();
+        RegexComparison regexComparison = new RegexComparison();
+
+        comparisons.put(equalToComparison.getFunctionName(), equalToComparison);
+        comparisons.put(regexComparison.getFunctionName(), regexComparison);
+
+        if (featuresList != null) {
+            ArrayList<FilterCondition> filterConditions = builder.getFilterConditions();
+
+            if (filterConditions.size() > 0) {
+                for (Feature feature : featuresList) {
+                    boolean skipFeature = false;
+                    if (builder.getSortProperty() != null
+                            && !feature.hasProperty(builder.getSortProperty())) {
+                        continue;
+                    }
+
+                    for (FilterCondition filterCondition : filterConditions) {
+                        if (feature.hasProperty(filterCondition.getPropertyName())) {
```

Comment:
I think you can define this`(filterCondition.getPropertyName()` once as it is used multiple times.